### ISSUE TITLE
Implement water boxes

### ISSUE THREAD

**Issue by jaburns** [libsm64/libsm64#2]

**Comment by theonerm2** [comment]
Any ideas on this one? I noticed at -100 on the y position of the world the player can swim. But how to implement a box of water? Changing Mario's action to swimming actions seems to do nothing unless he is below -100 on the y position of the world. I do have an idea. I'll see if it works.

**Comment by jaburns** [comment]
Yeah I'd just look at where the checks currently are that set Mario's state to swimming and replace them with a call to a new function that checks a list of water boxes. I'd be more than happy to accept a PR if you figure it out, I'm not gonna have time to work on this project myself for another couple months I think.
